components/script/dom/document.rs

```diff
@@ -2784,9 +2797,60 @@ impl Document {
             shadow_roots: DomRefCell::new(HashSet::new()),
             shadow_roots_styles_changed: Cell::new(false),
             media_controls: DomRefCell::new(HashMap::new()),
+            csp_list: DomRefCell::new(None),
+        }
+    }
+
+    pub fn set_csp_list(&self, csp_list: Option<CspList>) {
+        *self.csp_list.borrow_mut() = csp_list;
+    }
+
+    pub fn get_csp_list(&self) -> Option<Ref<CspList>> {
+        let b = self.csp_list.borrow();
+        if b.is_none() {
+            None
+        } else {
+            Some(Ref::map(b, |o| o.as_ref().unwrap()))
         }
     }
 
+    pub fn should_request_be_blocked_by_csp(&self, request: &RequestBuilder) -> csp::CheckResult {
```

Comment:
Spec link on this, and please add some in your crate too, having to jump from the code to the spec without them is making review more painful than it needs to be.